packages/core/spec/node/asciidoctor.spec.js

```diff
@@ -428,6 +428,33 @@ intro
         expect(errorMessage.message).to.equal('hi')
         expect(errorMessage.progname).to.equal('asciidoctor.js')
       })
+      it('should accept a message as block argument', () => {
+        const messages = []
+        const memoryLogger = asciidoctor.LoggerManager.newLogger('CustomMemoryLogger', {
+          add: function (severity, message, programName, block) {
+            if (typeof block === 'function') {
+              messages.push(block())
+            } else {
+              messages.push(message)
+            }
+          }
+        })
+        const defaultLogger = asciidoctor.LoggerManager.getLogger()
+        try {
+          asciidoctor.LoggerManager.setLogger(memoryLogger)
+          memoryLogger.add('error', 'test', 'asciidoctor.js')
+          asciidoctor.convert('Hello, {name}!', {
+            attributes: {
+              'attribute-missing': 'drop-line'
+            }
+          })
+          expect(messages.length).to.equal(2)
```

Comment:
I will also add a message after `convert` to make sure that the block is correctly removed.